chat-app/src/App.js

```diff
@@ -0,0 +1,177 @@
+
+export function App() {
+  const [username, setUsername] = useState('');
+  const [roomId, setRoomId] = useState('');
+  const [availableRooms, setAvailableRooms] = useState([]);
+  const [socket, setSocket] = useState(null);
+  const [connectionsCount, setConnectionsCount] = useState(0);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [roomName, setRoomName] = useState('');
+
+  const createRoom = async () => {
+    if (!roomName || roomName.trim() === '') {
+      setErrorMessage('Room name cannot be empty');
+      return;
+    }
+
+    try {
+      const response = await fetch('http://localhost:3005/rooms', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ roomName }),
+      });
+
+      const data = await response.json();
+      setAvailableRooms((prev) => {
+        const isRoomAlreadyExists = prev.some((room) => room.roomId === data.roomId);
+        if (!isRoomAlreadyExists) {
+          return [...prev, data];
+        }
+        return prev;
+      });
+      setRoomName('');
+      setRoomId(data.roomId);
+      setErrorMessage('');
+    } catch (err) {
+      console.error("Failed to create room:", err);
+    }
+  };
+
+  useEffect(() => {
+    const fetchRooms = async () => {
+      try {
+        const response = await fetch('http://localhost:3005/rooms');
+        const data = await response.json();
+        setAvailableRooms(data);
+      } catch (error) {
+        console.error('Failed to fetch rooms:', error);
+      }
+    };
+
+    fetchRooms();
+  }, []);
+
+  useEffect(() => {
+    const fetchConnectionsCount = async () => {
+      try {
+        const response = await fetch('http://localhost:3005/connections-count');
+        const data = await response.json();
+        setConnectionsCount(data.connectionsCount);
+      } catch (error) {
+        console.error('Failed to fetch connections count:', error);
+      }
+    };
+
+    fetchConnectionsCount();
+
+    const interval = setInterval(fetchConnectionsCount, 5000);
+
+    return () => clearInterval(interval);
+  }, []);
+
+  useEffect(() => {
+      console.log('Opening new WebSocket connection');
+      const ws = new WebSocket('ws://localhost:3005');
+      setSocket(ws);
+
+      ws.onopen = () => console.log('WebSocket connected');
+
+      ws.onmessage = (event) => {
+        const response = JSON.parse(event.data);
+
+        if (response.error) {
+          setErrorMessage(response.error);
+          console.error('Server error:', response.error);
+          return;
+        }
+
+        if (response.newRoom) {
+          setAvailableRooms((prev) => {
+            const exists = prev.some((room) => room.roomId === response.newRoom.roomId);
+            if (!exists) {
+              return [...prev, response.newRoom];
+            }
+            return prev;
+          });
+        }
+      };
+      ws.onclose = () => console.log('WebSocket closed');
```

Comment:
You might want to handle the WebSocket closure more gracefully by informing the user that the connection has been closed, especially if it's unexpected.